apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server.py

```diff
@@ -18,13 +18,123 @@
+    self.project_id = project_id
+    self.numeric_project_id = numeric_project_id
+    
+    self._credentials = {}
+    if use_default_credentials:
+      self._useDefaultCredentials()
+    if credentials_files:
+      self._useCredentials(credentials_files)
+
+    self._loadData()
+
+  def _useCredentials(self, credentials_files):
+    for path in string.split(credentials_files, os.pathsep):
+      account_credentials = service_account.ServiceAccountCredentials.from_json_keyfile_name(
+        path,
+        scopes = ServerState._credential_scopes)
+      self._credentials[account_credentials.service_account_email] = account_credentials
+
+  def _useDefaultCredentials(self):
+    self._credentials[u'default'] = service_account.ServiceAccountCredentials.get_application_default()
+
+  def _loadData(self):
+    with open(os.path.join(os.path.dirname(os.path.realpath(__file__)), 'initial_data.json')) as initial_data_file:
+      self._all_data = json.load(initial_data_file, object_pairs_hook=OrderedDict)
+      indexed_lists = [
+        u'instance/disks',
+        u'instance/licenses',
+        u'instance/network-interfaces',
+        u'instance/network-interfaces/*/access-configs',
+        u'instance/network-interfaces/*/forwarded-ips',
+        u'instance/network-interfaces/*/ip-aliases'
+      ]
+      for indexed_list in indexed_lists:
+        components = indexed_list.split('/')
+        def process(data, i):
+          component = components[i]
+          if component == '*':
+            for item in data:
+              process(item, i + 1)
+          elif i == len(components) - 1:
+            data[component] = MetadataIndexedList(data[component])
+          else:
+            process(data[component], i + 1)
+
+        process(self._all_data, 0)
+
+      # Replace service-accounts if any credentials were specified via flags.
+      if len(self._credentials) != 0:
+        def make_service_account(email, account_credentials):
+          is_default = email == u'default'
+          return OrderedDict((
+              (u'aliases', [u'default'] if is_default else []),
+              (u'email', account_credentials.client_id if is_default else account_credentials.service_account_email),
+              (u'scopes', ServerState._credential_scopes),
+              (u'token', None) # Real values will be substituted here upon request.
+            ))
+        self._all_data[u'instance'][u'service-accounts'] = OrderedDict(
+          {email : make_service_account(email, self._credentials[email]) for email in self._credentials})
+
+      numeric_project_id_token = u'<NUMERIC_PROJECT_ID>'
+      project_id_token = u'<PROJECT_ID>'
+      def convertToMetadata(value):
+        if isinstance(value, OrderedDict):
+          result = OrderedDict()
+          for key in value:
+            new_key = key.replace(numeric_project_id_token, unicode(self.numeric_project_id))
```

Comment:
Currently, there is no need for Unicode with the data used in initial_data.json. But I would like to add support for providing alternate initial data and/or updating data on the fly. I would like this to allow for Unicode values (in the instance description, for example). So I figured it was best to be consistent and use Unicode everywhere metadata values were stored. If you think this is unnecessary or there's a better way to do it in Python, please let me know.